apps/src/templates/studioHomepages/JoinSection.jsx

```diff
@@ -165,14 +209,25 @@ export default class JoinSection extends React.Component {
           />
           <Button
             __useDeprecatedTag
-            onClick={this.joinSection}
+            onClick={this.validateRecaptcha}
             className="ui-test-join-section"
             color={Button.ButtonColor.gray}
             disabled={this.state.sectionCode.length === 0}
             text={i18n.joinSection()}
             style={styles.button}
           />
         </div>
+        <div>
+          {isLoaded && requireCaptcha && (
+            <ReCaptchaDialog
+              isOpen={this.state.displayCaptcha}
```

Comment:
I'm a little confused as to why we have both `requireCaptcha` and `this.state.displayCaptcha`.  Are these two values always the same?

Reply:
`requireCaptcha` is whether or not the section attempts exceed 3. `displayCaptcha` is the boolean that represents whether or not the dialog is open. A user could have exceeded the 3 section attempt threshold, but that doesn't mean we want to automatically render the dialog on the homepage – that only happens after the user clicks the "Join Section" button. Does that make sense?